Sources/SourceKitLSP/Swift/SwiftLanguageServer.swift

```diff
@@ -419,11 +536,16 @@ extension SwiftLanguageServer {
         }
 
         req[keys.sourcetext] = edit.text
-
         lastResponse = try? self.sourcekitd.sendSync(req)
+      } afterCallback: { (after: DocumentSnapshot) in
+        if let dict = lastResponse {
+          return self.updatedLexicalAndSyntacticTokens(response: dict, for: after)
+        } else {
+          return nil
+        }
       }
 
-      if let dict = lastResponse, let snapshot = snapshot {
+      if let dict = lastResponse, let snapshot = self.documentManager.latestSnapshot(uri) {
```

Comment:
What’s the reason for not using the `snapshot` variable returned by `self.documentManager.edit` anymore? That one provided the guarantee that we are using the snapshot that resulted from the edit whereas a new snapshot might exist by the time that `latestSnapshot(uri)` is called.

Reply:
Ah thanks, I forgot to change this one back.